cdk/lib/sender-stack.ts

```diff
@@ -0,0 +1,145 @@
+import { GuAutoScalingGroup } from '@guardian/cdk/lib/constructs/autoscaling';
+import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
+import { AppIdentity, GuStack } from '@guardian/cdk/lib/constructs/core';
+import { GuVpc, SubnetType } from '@guardian/cdk/lib/constructs/ec2';
+import {
+	GuAllowPolicy,
+	GuInstanceRole,
+} from '@guardian/cdk/lib/constructs/iam';
+import type { GuAsgCapacity } from '@guardian/cdk/lib/types';
+import type { App } from 'aws-cdk-lib';
+import { CfnOutput, Duration } from 'aws-cdk-lib';
+import { HealthCheck, ScalingEvents } from 'aws-cdk-lib/aws-autoscaling';
+import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { Queue } from 'aws-cdk-lib/aws-sqs';
+import {
+	ParameterDataType,
+	ParameterTier,
+	StringParameter,
+} from 'aws-cdk-lib/aws-ssm';
+
+interface SenderStackProps extends GuStackProps {
+	appName: string;
+	asgCapacity: GuAsgCapacity;
+	instanceType: InstanceType;
+	notificationSnsTopic: string;
+	cleanerQueueArn: string;
+}
+
+export class SenderWorkerStack extends GuStack {
+	constructor(scope: App, id: string, props: SenderStackProps) {
+		super(scope, id, props);
+
+		const sqsMessageVisibilityTimeout = Duration.seconds(100);
+		const sqsMessageRetentionPeriod = Duration.hours(1);
+		const sqsMessageRetryCount = 5;
+		const instanceType = InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL);
+		const targetUtilizationPercent = 40;
```

Comment:
I don't have a feel for what this number should be. Did we choose this based on your experience or some advice?

Reply:
Sorry I should give some explanation.

`sqsMessageRetentionPeriod` refers to the duration for how long a message can stay in the queue.  After this period elapses, the message get deleted even if it is not processed.

`sqsMessageRetryCount` refers to the number of times a message can get picked up before being sent to DLQ.  So it can be seen as the maximum retry count.

For now, it seems sensible to use the same values for the old queues, but we can review them later.

For `instanceType`, I think we can start with a small box size that is reasonable.  `t4g.small` is chosen because it is used by Apps-rendering on `PROD`.  But definitely we can get a better idea what this value should be after we implement the sender and run some testing.

For `targetUtilizationPercent`, it defines the level of CPU utilisation at which the scaling policy scales up or down the autoscaling group.  You have raised a good point here.  `t4g.small` is a burstable instance class and its baseline performance is 20%.  Going over this usage consumes the CPU credit.  We should probably set the target CPU utilisation to 20.  (And it is the same value as in MAPI).

Please let me know if you have any comments.

Reply:
Ah, the target utilisation makes sense now, thank you :) Given how aws burst the instance, changing this value to 20 seems reasonable!